eqcorrscan/utils/pre_processing.py

```diff
@@ -846,10 +846,39 @@ def _prep_data_for_correlation(stream, templates, template_names=None,
     # Remove templates with no matching channels
     filt = np.ones(len(template_names)).astype(bool)
     for i, template in enumerate(templates):
-        template_ids = {tr.id for tr in template}
-        if len(template_ids.intersection(stream_ids)) == 0:
+        trace_ids = {tr.id for tr in template}
+        if len(trace_ids.intersection(stream_ids)) == 0:
             filt[i] = 0
 
+    # Ensure that the templates' earliest traces are kept, even if there is no
+    # continuous data for them. If this happens, we need to add a NaN-stream to
+    # the continuous data to avoid inconsistent detection times.
+    n_template_traces = np.array([len(temp) for temp in templates])
+    n_stream_traces = sum([n+1 for s, n in seed_ids])
+    # These checks are not necessary if all templates will get NaN-traces,
+    # because the NaN-traces will save the right starttime for the template.
+    nan_stream_ids = list()
+    if any(n_template_traces > n_stream_traces):
+        earliest_templ_trace_ids = list(set(
```

Comment:
May not need to convert from set to list.

Reply:
True; don't need to convert to list.